### PR TITLE
Update mocha to 2.2.5

### PR DESCRIPTION
**Pull request by hatched**
The up to date version of mocha is considerably faster and less buggy than the very old version that we were still using.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

As long as CI passes it should be ok but you may want to run it locally after a clean-all.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2823/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2825/
Test FAILed.

**Comment by hatched** [comment]
When run manually the CI passed successfully: http://ci.jujugui.org:8080/job/juju-gui/2826/console

File: jujugui/static/gui/src/test/test_viewlet_manager.js [hunk 1]
```diff
@@ -100,6 +100,9 @@ describe('Viewlet Manager', function() {
     });
   });
 
+  // Required so that the cleanups instance can be attached.
+  beforeEach(function() {});
```
Comment (jaycee) [inline jujugui/static/gui/src/test/test_viewlet_manager.js]: So is this a test-by-test situation, or do all test cases now need before/afterEach functions?

**Comment by jaycee** [comment]
Code is :+1: with one question. QA shows tests running about as reliably they did before. So QA OK, but also QA :cry:

**Comment by anthonydillon** [comment]
Code is good but tests still failing :+1:

**Comment by hatched** [comment]
Thanks for the reviews! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**